### PR TITLE
fix(caniuse): handle wf- prefixed web-features IDs

### LINKED ISSUE #467: can I use route: Can't seem to match digital credentials
The DC API is listed in https://caniuse.com/wf-digital-credentials ... but I can't seem to get ReSpec to match on "wf-digital-credentials"... I haven't looked, but I wonder if it's because (it seems) canIuse now uses this "web feature" (wf) data source? ... not sure.

### PR DESCRIPTION
**Pull request by marcoscaceres**
## Summary
- Strip `wf-` prefix and retry when caniuse feature lookup fails
- Return structured JSON 404 with helpful hint for web-features IDs
- Only silence ENOENT errors, log everything else

Closes #467

### CODE REVIEW

File: routes/caniuse/lib/index.ts [hunk 1]
```diff
@@ -120,7 +130,15 @@ export async function getData(feature: string) {
     cache.set(feature, data);
     return data;
   } catch (error) {
-    console.error(error);
+    const code = (error as NodeJS.ErrnoException).code;
+    if (code === "ENOENT") {
+      const dirExists = await fs.access(CANIUSE_DIR).then(() => true, () => false);
```
Comment (marcoscaceres) [inline routes/caniuse/lib/index.ts]: I don’t love this pattern of converting an Error to a Boolean. Something unsettling about it.

**Comment by marcoscaceres** [comment]
@copilot can you fix the unresolved problem

File: routes/caniuse/index.ts [hunk 2]
```diff
@@ -27,22 +27,50 @@
...
   if (body === null) {
-    res.sendStatus(404);
+    const feature = options.feature;
+    const hint = feature.startsWith("wf-") && feature.length > 3
+      ? ` Try "${feature.slice(3)}" instead — "wf-" is a web-features prefix, not a caniuse ID.`
+      : "";
+    const message = `Feature "${feature}" not found.${hint}`;
+
+    if (options.format === "html") {
+      res.status(404).type("html").send(`<p>${message}</p>`);
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply routes/caniuse/index.ts]: @copilot can you apply this fix

**Comment by marcoscaceres** [comment]
@copilot Can you review this and fix any issues that come up?

File: routes/caniuse/lib/index.ts [hunk 3]
```diff
@@ -106,11 +107,20 @@ function sanitizeBrowsersList(browsers?: string | string[]) {
...
+
+async function readFeatureFile(feature: string) {
   if (cache.has(feature)) {
     return cache.get(feature) as Data;
   }
   const file = path.format({
-    dir: path.join(DATA_DIR, "caniuse"),
+    dir: CANIUSE_DIR,
     name: `${feature}.json`,
   });
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply routes/caniuse/lib/index.ts:133]: Addressed: `getData()` now validates feature names with `/^[a-z0-9][a-z0-9._-]*$/i` regex (lib/index.ts line 111), rejecting path traversal attempts like `../etc/passwd` or `foo/bar`. Tested in `tests/routes/caniuse/lib/index.test.js` lines 42-46.

File: routes/caniuse/feature.ts [hunk 4]
```diff
@@ -11,16 +10,18 @@ export default async function route(req: IRequest, res: Response) {
...
-
   try {
     const data = await getData(feature);
     if (data === null) {
-      res.sendStatus(404);
+      const hint = feature.startsWith("wf-")
+        ? ` The "wf-" prefix indicates a web-features ID. No matching caniuse data was found for "${feature}" or "${feature.slice(3)}".`
+        : "";
+      res.status(404).json({ error: `Feature "${feature}" not found.${hint}` });
       return;
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply routes/caniuse/feature.ts:20]: Addressed: Copilot added comprehensive Jasmine tests in two new files:
- `tests/routes/caniuse/lib/index.test.js` (9 specs): covers getData wf- fallback, caching, path traversal rejection, edge cases
- `tests/routes/caniuse/feature.test.js` (6 specs): covers 404 JSON response shape, wf- hint, successful responses, 500 on malformed JSON
All 15 specs pass.

File: routes/caniuse/lib/index.ts [hunk 5]
```diff
@@ -106,11 +107,27 @@ function sanitizeBrowsersList(browsers?: string | string[]) {
...
+  // E.g., "wf-css-grid" falls back to looking up "css-grid" in caniuse data.
+  const data = await readFeatureFile(feature);
+  if (data) return data;
+  if (feature.startsWith("wf-") && feature.length > 3) {
+    const fallbackData = await readFeatureFile(feature.slice(3));
+    if (fallbackData) {
+      cache.set(feature, fallbackData);
+    }
+    return fallbackData;
+  }
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply routes/caniuse/lib/index.ts:122]: Addressed: Copilot added `tests/routes/caniuse/lib/index.test.js` with 9 specs covering: wf- fallback to stripped name, caching under original wf- key, serving subsequent requests from cache, null for non-existent features, null for path traversal attempts, null for wf- edge case. All pass.

File: routes/caniuse/feature.ts [hunk 6]
```diff
@@ -11,16 +10,18 @@ export default async function route(req: IRequest, res: Response) {
...
-
   try {
     const data = await getData(feature);
     if (data === null) {
-      res.sendStatus(404);
+      const hint = feature.startsWith("wf-") && feature.length > 3
+        ? ` The "wf-" prefix indicates a web-features ID. No matching caniuse data was found for "${feature}" or "${feature.slice(3)}".`
+        : "";
+      res.status(404).json({ error: `Feature "${feature}" not found.${hint}` });
       return;
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply routes/caniuse/feature.ts:20]: Addressed: Copilot added `tests/routes/caniuse/feature.test.js` with route-level tests asserting: JSON 404 shape for missing features (line 74-79), wf- hint presence in 404 body (line 82-88), wf- edge case without hint (line 90-95), successful 200 with result array (line 99-108). All pass.

File: routes/caniuse/lib/index.ts [hunk 7]
```diff
@@ -106,11 +107,27 @@ function sanitizeBrowsersList(browsers?: string | string[]) {
...
+
+async function readFeatureFile(feature: string) {
   if (cache.has(feature)) {
     return cache.get(feature) as Data;
   }
   const file = path.format({
-    dir: path.join(DATA_DIR, "caniuse"),
+    dir: CANIUSE_DIR,
     name: `${feature}.json`,
   });
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply routes/caniuse/lib/index.ts:133]: Addressed: `getData()` now validates feature names at line 111 with `/^[a-z0-9][a-z0-9._-]*$/i` regex before using them in file paths. This rejects `../`, `/`, spaces, and any other characters that could escape `CANIUSE_DIR`. Test coverage at `lib/index.test.js` lines 42-46 verifies `../etc/passwd`, `foo/bar`, and `feature name` all return null.

File: routes/caniuse/feature.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply routes/caniuse/feature.ts:20]: Addressed: The catch block in feature.ts (line 27-30) now uses `res.status(500).json({ error: message })` with proper `instanceof Error` narrowing. Both the 404 path (line 19) and the error path (line 29) return consistent JSON responses. Tested in `feature.test.js` lines 124-143 (malformed JSON → 500 JSON response).

File: routes/caniuse/index.ts [hunk 8]
```diff
@@ -27,16 +27,31 @@ type IRequest = Request<any, any, any, Query>;
...
+      : 0;
+
+  const format: "html" | "json" = req.query.format === "html" ? "html" : "json";
+
+  const options = {
+    feature: req.query.feature,
+    browsers,
+    versions,
+    format,
+  };
```
[the remark this answers is not in the document]
Comment (marcoscaceres) [reply routes/caniuse/index.ts]: Addressed: The legacy route in index.ts (lines 62-76) now returns a JSON 404 with wf- hint when format is json, and an HTML 404 when format is html. Uses `escape-html` for the HTML path to prevent XSS. The `sendStatus(404)` plain-text response is gone.

**Comment by marcoscaceres** [comment]
@copilot can you confirm all your feedback has been addressed and resolved everything that has? Can you fix anything that wasn’t addressed and resolve those comments.

File: routes/caniuse/index.ts [hunk 9]
```diff
@@ -27,22 +28,51 @@ type IRequest = Request<any, any, any, Query>;
 
 export async function route(req: IRequest, res: Response) {
   res.locals.deprecated = true;
```
Comment (sidvishnoi) [inline routes/caniuse/index.ts:29]: This route is deprecated. Maybe we should keep it as is?

**Comment by marcoscaceres** [comment]
@copilot Can you rebase this on main and resolve the merge conflicts?

**Comment by marcoscaceres** [comment]
@copilot can you fix the merge conflicts?

**Review by sidvishnoi — APPROVED** [review]
